dask_planner/src/sql.rs

```diff
@@ -140,12 +140,21 @@ impl ContextProvider for DaskSQLContext {
         }
 
         // Loop through all of the user defined functions
-        for schema in self.schemas.values() {
+        let schemas = self.schemas.clone();
+        for schema in schemas.values() {
             for (fun_name, function) in &schema.functions {
                 if fun_name.eq(name) {
                     let sig = Signature::variadic(vec![DataType::Int64], Volatility::Immutable);
```

Comment:
Is it intentional that we are declaring all of the functions to accept a single parameter of type `Int64` ?

Reply:
Think that's just an artifact of copying the code from the above function meta definitions - I think what we would want here is to take the input types stored as the keys of ``function.return_types` and make multiple `TypeSignature::Exacts` that are then compiled in a `Signature::one_of`.

This still leaves something to be desire though, as IIUC this means that if we passed through inputs that didn't match any signatures, but could be coerced into matching input (e.g. we pass `Int64` columns into a function defined for `Float64`), we will error out, where the Calcite branch would do this coercion.

Reply:
I would expect DataFusion to also coerce the types.